python/pyspark/ml/pipeline.py

```diff
@@ -242,7 +242,7 @@ def load(self, path):
             return JavaMLReader(self.cls).load(path)
         else:
             uid, stages = PipelineSharedReadWrite.load(metadata, self.sc, path)
-            return PipelineModel(stages=stages)._resetUid(uid)
+            return self.cls(stages=stages)._resetUid(uid)
```

Comment:
I don't know a lot of ML stuff but I can tell the current PR is inconsistent with Scala side:

https://github.com/apache/spark/blob/3ce1dff7ba4799295a2bf89e6df35e900afc587b/mllib/src/main/scala/org/apache/spark/ml/Pipeline.scala#L362

Is `PipelineModel` supposed to be inherited?  If that's the case seems we should also inherit `PipelineModelReader` and `PipelineModelWriter` accordingly consistently with Scala and Java side.

Reply:
Is that mandatory? It seems strange that the intention is to force developers to inherit all three together even when the Reader/Writer logic is the same for the custom implementation of the `PipelineModel`.

Reply:
We should reduce such difference in other languages, for example, in order to reduce maintenance overhead. Unless there's a strong reason, I think we don't usually make difference between them.

It is by design - how/which scope the project wants to expose. If we should make it inheritable, ideally we should change Scala side too.

From my reading of the codes, I don't think it was intended to expose and make it inheritable, though.

Reply:
I see. OK so should I just go ahead and close the pull request?

Reply:
Yes, I think we should just close but I'd wait for a couple of days in case other committers have different thoughts in particular given I only know limited context of ML.